Add new sensor platform for Islamic prayer times

uchagani opened the pull request:
## Description:

Adds a new sensor platform for Islamic prayer times.

**Pull request in [home-assistant.io](https://github.com/home-assistant/home-assistant.io) with documentation (if applicable):** https://github.com/home-assistant/home-assistant.io/pull/7783

## Example entry for `configuration.yaml` (if applicable):
```yaml
sensor:
  - platform: islamic_prayer_times
```

## Checklist:
  - [x] The code change is tested and works locally.
  - [x] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**
  - [x] There is no commented out code in this PR.

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [home-assistant.io](https://github.com/home-assistant/home-assistant.io)

If the code communicates with devices, web services, or third-party tools:
  - [x] New dependencies have been added to the `REQUIREMENTS` variable ([example][ex-requir]).
  - [x] New dependencies are only imported inside functions that use them ([example][ex-import]).
  - [x] New or updated dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`.
  - [x] New files were added to `.coveragerc`.

If the code does not interact with devices:
  - [x] Tests have been added to verify that the new code works.

[ex-requir]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L14
[ex-import]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L54

balloob committed:
Bumped version to 0.84.0b0

balloob committed:
Revert #17745 (#19064)

mikeage committed:
Fix missing colorTemperatureInKelvin from Alexa responses (#19069)

* Fix missing colorTemperatureInKelvin from Alexa responses

* Update smart_home.py

* Add test

amelchio committed:
Upgrade aiolifx to 0.6.7 (#19077)

bramkragten committed:
Force refresh Lovelace (#19073)

* Force refresh Lovelace

* Check config on load

* Update __init__.py

* Update __init__.py

balloob committed:
Updated frontend to 20181207.0

balloob committed:
Bumped version to 0.84.0b1

marchingphoenix committed:
home-assistant/home-assistant#18645: Fix climate mode mapping.

marchingphoenix committed:
home-assistant/home-assistant#18645: Fix climate mode mapping.

marchingphoenix committed:
home-assistant/home-assistant#18645: Remove un-used constants.

marchingphoenix committed:
home-assistant/home-assistant#18645: Remove un-used constants.

postlund committed:
Upgrade pyatv to 0.3.12 (#19085)

marchingphoenix committed:
home-assistant/home-assistant#18645: revert heat-cool -> auto change

marchingphoenix committed:
home-assistant/home-assistant#18645: revert heat-cool -> auto change

nhorvath committed:
Bump skybellpy version to fix api issue (#19100)

Danielhiversen committed:
Upgrade Tibber lib (#19098)

syssi committed:
Don't avoid async_schedule_update_ha_state by returning false (#19102)

arigilder committed:
Remove marking device tracker stale if state is stale (#19133)

Commit:
add new sensor platform for Islamic prayer times

Commit:
add new sensor platform for Islamic prayer times

Commit:
fix datetime issue

Commit:
fix datetime issue

Commit:
update tests

Commit:
update tests

basschipper committed:
Fixed doorbird config without events (empty list) (#19121)

RyuzakiKK committed:
Add code support for iAlarm (#19124)

fabaff committed:
Upgrade slacker to 0.12.0

fabaff committed:
Upgrade youtube_dl to 2018.12.03 (#19139)

Commit:
fix linting errors

Commit:
fix linting errors

Commit:
fix linting errors

Commit:
fix linting errors

Commit:
fix linting errors

Commit:
fix linting errors

scop committed:
Upgrade upcloud-api to 0.4.3

scop committed:
Upgrade mypy to 0.650 (#19150)

* Upgrade to 0.650

* Remove no longer needed type: ignore

Commit:
update tests to use @MockDependency

Commit:
update tests to use @MockDependency

tinloaf committed:
Update geizhals dependency (#19152)

fabaff committed:
Upgrade sphinx-autodoc-typehints to 1.5.2 (#19140)

fabaff committed:
Upgrade slacker to 0.12.0 (#19142)

fabaff committed:
Merge pull request #19148 from scop/upgrade-upcloud

Upgrade upcloud-api to 0.4.3

craftyguy committed:
Update radiotherm to 2.0.0 and handle change in tstat error detection (#19107)

* radiotherm: bump version to 2.0.0

* radiotherm: change handling of transient errors from tstat

Radiotherm 2.0.0 now throws an exception when a transient error is
detected, instead of returning -1 for the field where the error was
detected. This change supports handling the exception.

jarondl committed:
Update pygtfs to upstream's 0.1.5 (#19151)

This works by adding `?check_same_thread=False` to the sqlite
connection string, as suggested by robbiet480@.
It upgrades pygtfs from homeassitant's forked 0.1.3 version to 0.1.5.

Fixes #15725

balloob committed:
Lovelace using storage (#19101)

* Add MVP

* Remove unused code

* Fix

* Add force back

* Fix tests

* Storage keyed

* Error out when storage doesnt find config

* Use old load_yaml

* Set config for panel correct

* Use instance cache var

* Make config option

balloob committed:
Lovelace using storage (#19101)

* Add MVP

* Remove unused code

* Fix

* Add force back

* Fix tests

* Storage keyed

* Error out when storage doesnt find config

* Use old load_yaml

* Set config for panel correct

* Use instance cache var

* Make config option

balloob committed:
Updated frontend to 20181210.0

balloob committed:
Updated frontend to 20181210.0

balloob committed:
Update translations

balloob committed:
Bumped version to 0.84.0b2

balloob committed:
Drop OwnTracks bad packets (#19161)

balloob committed:
Drop OwnTracks bad packets (#19161)

balloob committed:
Fix lovelace save (#19162)

balloob committed:
Fix lovelace save (#19162)

marchingphoenix committed:
Google assistant fix target temp for *F values. (#19083)

* home-assistant/home-assistant#18524 : Add rounding to *F temps

* home-assistant/home-assistant#18524 : Linting

* simplify round behavior

* fix trailing whitespace

(thanks github editor)

marchingphoenix committed:
Google assistant fix target temp for *F values. (#19083)

* home-assistant/home-assistant#18524 : Add rounding to *F temps

* home-assistant/home-assistant#18524 : Linting

* simplify round behavior

* fix trailing whitespace

(thanks github editor)

balloob committed:
Google_assistant: fix climate modes (#19084)

* home-assistant/home-assistant#18645: Fix climate mode mapping.

* home-assistant/home-assistant#18645: Remove un-used constants.

* home-assistant/home-assistant#18645: revert heat-cool -> auto change

balloob committed:
Updated frontend to 20181210.1

balloob committed:
Updated frontend to 20181210.1

balloob committed:
Add raw service data to event (#19163)

balloob committed:
Add raw service data to event (#19163)

balloob committed:
Bumped version to 0.84.0b3

dmulcahey committed:
ZHA - Event foundation (#19095)

* event foundation

* add missing periods to comments

* reworked so that entities don't fire events

* lint

* review comments

fredrike committed:
TelldusLive config flow (#18758)

* update TelldusLive to use config flow

* fixes from Martin

* Update homeassistant/components/tellduslive/config_flow.py

Co-Authored-By: fredrike <fredrik.e@gmail.com>

* revert changes in entry.py

* tox tests

* tox fixes

* woof woof (fix for hound)

* lint ignore

* unload entry

* coverall toxtests

* fix some toxtests

balloob committed:
Fix cloud defaults (#19172)

balloob committed:
Fix cloud defaults (#19172)

balloob committed:
Updated frontend to 20181211.0

balloob committed:
Updated frontend to 20181211.0

balloob committed:
Bumped version to 0.84.0b4

jkeljo committed:
Enable alarmdecoder to see open/close state of bypassed RF zones when armed (#18477)

* Enable alarmdecoder to see open/close state of bypassed zones when armed

The alarmdecoder component already reported RF state bits as attributes. If the user knows which loop is set up for the zone in the alarm panel, they can use that information to tell whether the zone is open or closed even when the system is armed by monitoring the appropriate attribute. That’s awkward, so this commit enables the user to simply configure which loop is used and the component will update the state itself.

* Simplify, also it's more correct to treat it as a state change rather than a
permanent state, since it's possible the decoder might miss some events.

* Remove relative import

fabaff committed:
Add SCAN_INTERVAL (#19186)

* Add SCAN_INTERVAL

* More clean-up

emontnemery committed:
Fail if new entity_id is in hass.states

emontnemery committed:
Fail if new entity_id is in hass.states

javicalle committed:
Restore states for RFLink devices (#18816)

* Merge branch 'master' of https://github.com/home-assistant/home-assistant into dev

# Conflicts:
#	homeassistant/components/binary_sensor/point.py
#	homeassistant/components/cloud/__init__.py
#	homeassistant/components/cloud/prefs.py
#	homeassistant/components/frontend/__init__.py
#	homeassistant/components/light/fibaro.py
#	homeassistant/components/logbook.py
#	homeassistant/components/point/__init__.py
#	homeassistant/config_entries.py
#	homeassistant/const.py
#	homeassistant/helpers/service.py
#	requirements_all.txt
#	requirements_test_all.txt

* one 'async_get_last_state' refactor left behind

* Remove RestoreEntity inheritance (already in parent class)

* # pylint: disable=too-many-ancestors

* code predictor can be a bitch

* lint corrections

* # pylint: disable=too-many-ancestors

* recover from dict[key]

* Remove all 'coroutine' decorator, replace for 'async def'
Replace all 'yield from' for 'await'
Replace 'hass.async_add_job' for 'hass.async_create_task'

fredrike committed:
Move daikin to package (#19187)

emontnemery committed:
Move check to websocket

emontnemery committed:
Move check to websocket

Danielhiversen committed:
Update switchbot library (#19202)

fredrike committed:
add unique_id to SMHI (#19183)

balloob committed:
Revert PR #18602 (#19188)

balloob committed:
Revert PR #18602 (#19188)

balloob committed:
Bumped version to 0.84.0

balloob committed:
Merge pull request #19215 from home-assistant/rc

0.84

hunterjm committed:
Add automation and script events to logbook event types (#19219)

hunterjm committed:
Add automation and script events to logbook event types (#19219)

tinloaf committed:
Fix geizhals crash if no price found (#19197)

* Fix geizhals crash if no price found

* Return None on unknown price.

* Linting

* Linting the linting

emontnemery committed:
Add JSON attribute topic to MQTT sensor

emontnemery committed:
Review comments

emontnemery committed:
Review comments

dmulcahey committed:
Add ZHA device handler library (#19099)

* event foundation

* implement quirks

* lock zha-quirks version

* allow quirks handling to be toggled on and off

* revert event commit

* disable warning

* update requirements_all

* Remove fix in favor of #19141

#19141 should be what ultimately corrects this issue.

* review comment

balloob committed:
Fix owntracks topic in encrypted ios (#19220)

* Fix owntracks topic

* Warn if per-topic secret and using HTTP

balloob committed:
Fix owntracks topic in encrypted ios (#19220)

* Fix owntracks topic

* Warn if per-topic secret and using HTTP

balloob committed:
Bumped version to 0.84.1

balloob committed:
Merge pull request #19228 from home-assistant/rc

0.84.1

balloob committed:
Merge remote-tracking branch 'origin/master' into dev

bachya committed:
Add package data attribute to 17track.net summary sensors (#19213)

* 17track.net: Add package data attribute to summary sensors

* Member comments

kennedyshead committed:
Bump aioasuswrt (#19229)

* bump aioasuswrt version

* run gen_requirements

kennedyshead committed:
Bump aioasuswrt (#19229)

* bump aioasuswrt version

* run gen_requirements

rytilahti committed:
Convert songpal to use asynchronous websocket for state updates (#19129)

* Add websocket-based non-polling variant for songpal

* linting fixes

* changes based on Martin's feedback

* Fix linting

* add backoff timer for reconnects, fix variable naming (I thought that this wouldn't matter for internals..)

* Remove poll configuration variable

* bump the version just to be sure, the previous release lacked a version file (required for setup.py)

foxel committed:
Bumped NDMS2 client to 0.0.6 (#19244)

balloob committed:
RFC: Deprecate auto target all for services and introduce entity_id: * (#19006)

* Deprecate auto target all

* Match on word 'all'

balloob committed:
Make automations log errors (#18965)

molobrakos committed:
Re-use connection-pool (#19249)

Re-use connection-pool of VOC

balloob committed:
Check admin permission before able to manage config entries

emontnemery committed:
Add device registry to MQTT light (#19013)

Devqon committed:
Support for the Harman Kardon AVR (#18471)

* Feature: support for the HK AVR

* Remove testcode

* Feature: support for the HK AVR

* Remove testcode

* Added checklist

* Review fixes whitespaces

* Lint fixes

* Review fixes, add current source

* Remove unused imports

* Review fixes; State constants, dict[key]

* More review fixes, Unknown state and Sources

* Review fix; rename devices to entities

fredrike committed:
Fix point sensor discovery (#19245)

fabaff committed:
Fix list (fixes #19235) (#19258)

fabaff committed:
Fix list (fixes #19235) (#19258)

teharris1 committed:
Resolve IOLinc sensor name (#19050)

balloob committed:
Lint

balloob committed:
Fix OwnTracks deadlocking (#19260)

* Fix OwnTracks deadlocking

* Fix deadlock

balloob committed:
Fix OwnTracks deadlocking (#19260)

* Fix OwnTracks deadlocking

* Fix deadlock

balloob committed:
Improve check for duplicated entity_id (#19194)

* Fail if new entity_id is in hass.states

* Move check to websocket

* Review comments

damarco committed:
Always add friendly name attribute to ZHA entities (#19141)

* Always add friendly name attribute

* Only change device_info name

damarco committed:
Always add friendly name attribute to ZHA entities (#19141)

* Always add friendly name attribute

* Only change device_info name

rohankapoorcom committed:
Expose ZoneMinder availability to Home Assistant (#18946)

* Expose ZoneMinder availability to Home Assistant

* Bump zm-py to 0.2.0 with the availability changes published

hfurubotten committed:
Upgraded enturclient to 0.1.2 (#19267)

bremor committed:
Synology chat add verify ssl (#19276)

* Update synology_chat.py

* Added verify_ssl option to notify.synology_chat

Python requests will verify ssl by default, this configuration options allows the user to specify if they want to verify ssl or not. Non breaking change, default is True - do verify ssl.

marchingphoenix committed:
Fix call to super() (#19279)

* home-assistant/home-assistant#19273: fix call to super()

* home-assistant/home-assistant#19273: adjust to python3 standards.

* home-assistant/home-assistant#19273: remove bad test.

marchingphoenix committed:
Fix call to super() (#19279)

* home-assistant/home-assistant#19273: fix call to super()

* home-assistant/home-assistant#19273: adjust to python3 standards.

* home-assistant/home-assistant#19273: remove bad test.

thibmaek committed:
Add NMBS (Belgian railway) sensor platform (#18610)

* Add custom component to core

* Add pyrail to reqs

* Format & lint

* Sort nmbs.py into place on coveragerc

* Only set up station live if provided

* Only set up sensor if config is provided

* Fix line too long linting error

* PR Remarks

* Add docstrings

* Fix hound line to long error

* Fix quotes

* Rebase coveragerc

* PR Review

* Init empty attrs

* Dont include delay if there is none

* PR review

* Safer check

* Rebase reqs

* Generate req

* Update homeassistant/components/sensor/nmbs.py

Co-Authored-By: thibmaek <thibault.maekelbergh@iCloud.com>

* PR remarks

balloob committed:
Add permission check to light service (#19259)

balloob committed:
Check admin permission before able to manage config entries (#19265)

* Check admin permission before able to manage config entries

* Lint

balloob committed:
Rename is_owner decorator to is_admin (#19266)

* Rename is_owner decorator to is_admin

* Update test_auth.py

luca-angemi committed:
Add automation and script events to logbook filter events (#19253)

* Add automation and script events to logbook filter events

* Update logbook.py

* Update logbook.py

* Update logbook tests

* Update test_logbook.py

* Update test_logbook.py

* Update test_logbook.py

* Update test_logbook.py

luca-angemi committed:
Add automation and script events to logbook filter events (#19253)

* Add automation and script events to logbook filter events

* Update logbook.py

* Update logbook.py

* Update logbook tests

* Update test_logbook.py

* Update test_logbook.py

* Update test_logbook.py

* Update test_logbook.py

balloob committed:
Bumped version to 0.84.2

emontnemery committed:
Fix race in entity_platform.async_add_entities (#19222)

emontnemery committed:
Fix race in entity_platform.async_add_entities (#19222)

balloob committed:
Merge pull request #19281 from home-assistant/rc

0.84.2

balloob committed:
Merge branch 'master' into dev

marchingphoenix committed:
home-assistant/home-assistant#17333: update to use DOMAIN constants and standards. (#19242)

emontnemery committed:
Make variable `entity_id` available to value_template for MQTT binary sensor (#19195)

* MQTT binary_sensor: Make variable `entity_id` available to value_template

* Review comments

* Add testcase

MaxG88 committed:
Set unavailable when unreachable (#19012)

* Turn GPMDP Off When Unavailable

* Update requirements_all.txt

* Specified Exception Type

* Update gpmdp.py

molobrakos committed:
Updated ELIQ Online sensor to async API (#19248)

* Updated ELIQ Online sensor to async API

* Remove use of STATE_UNKNOWN

fabaff committed:
Add air pollutants component (#18707)

* Add air pollutants component

* Fix lint issue

* Update docstrings

* Revert change

* Remove entries

* Remove imports

* Fix variable and other fixes

* Change tests

* Set SCAN_INTERVAL

pbalogh77 committed:
Device config for Fibaro hub integration (#19171)

wonderslug committed:
Set InsteonEntity name to be combo of description and address. (#17262)

* Set InsteonEntity name to be combo of description and address.

ie "LampLinc Dimmer 26453a" or "Keypad Dimmer 291abb_2"

Using a centralized network name

* Updated the name to have a more contextual references for device
functions then just the group id.

* Cleanup for hound

* Removed the _generate_network_address function.  Not used anymore

* Cleanup for lint

* clean for hound

* Moved descriptor mapper to be a class variable of the InsteonEntity in order to fix lib loading issue for lint

* Well, moved DescriptorMapper instance to a function variable now...

* fix hound

* Lint Cleanup

* Clean up docstrings

* Simplify Label lookup based on state name

liaanvdm committed:
Fix restore state for manual alarm control panel (#19284)

* Fixed manual alarm control panel restore state

* Revert "Fixed manual alarm control panel restore state"

This reverts commit 61c9faf434a8bb276133578a0811100a796784ca.

* Fixed manual alarm control panel's state restore

ColinHarrington committed:
Adding support for Plum Lightpad (#16576)

* Adding basic Plum Lightpad support - https://plumlife.com/

* Used Const values
is_on is a bool

* Added LightpadPowerMeter Sensor to the plum_lightpad platform

* Moved to async setup, Introduced a PlumManager, events, subscription, Light and Power meter working

* Added PlumMotionSensor

* Added Glow Ring support

* Updated plum library and re-normalized

* set the glow-ring's icon

* Naming the glow ring

* Formatting and linting

* Cleaned up a number of linting issues.  Left a number of documentation warnings

* setup_platform migrated to async_setup_platform Plum discovery run as a job

* bumped plumlightpad version

* On shutdown disconnect the telnet session from each plum lightpad

* Cleanup & formatting. Worked on parallell cloud update

* Moved setup from async to non-async

* Utilize async_call_later from the helpers

* Cleanedup and linted, down to documentation & one #TODO

* Remove commented out debug lines

* Fixed Linting issues

* Remove TODO

* Updated comments & fixed Linting issues

* Added plumlightpad to requirements_all.txt

* Fixing imports with isort

* Added components to .coveragerc

* Added PLUM_DATA constant for accessing hass.data[PLUM_DATA]

* used dictionary syntax vs get(...) for config

* Linting needed an additonal line

* Fully async_setup now. removed @callback utilize bus events for detecting new devices found.

* Upgraded to plumlightpad 0.0.10

* Removed extra unused PLATFORM_SCHEMA declarations

* Moved listener attachment to `async_added_to_hass` and removed unused properties & device_state_attributes

* Utilized Discovery when devices were located

* Linting and cleanup

* used `hass.async_create_task` instead of `hass.async_add_job` per Martin

* Removed redundant criteria in if block

* Without discovery info, there is no need to setup

* Better state management and async on/off for Glow Ring

* renamed async_set_config back to set_config, fixed cleanup callback and Plum Initialization

* Fixed flake8 linting issues

* plumlightpad package update

* Add 'motion' device_class to Motion Sensor

* Fixed last known Linting issue

* let homeassistant handle setting the brightness state

* String formatting vs concatenation

* use shared aiohttp session from homeassistant

* Updating to use new formatting style

* looks like @cleanup isn't neccesary

* ditch the serial awaits

* Ensure async_add_entities is only called once per async_setup_platform

* Creating tasks to wait for vs coroutines

* Remove unused white component in the GlowRing

* Used local variables for GlowRing colors & added a setter for the hs_color property to keep the values in sync

* Linted and added docstring

* Update the documentation path to point to the component page

* Removed the extra sensor and binary_sensor platforms as requested. (To be added in later PRs)

* Update plum_lightpad.py

* Update plum_lightpad.py

scop committed:
huawei_lte: Fetch only required data (#17618)

When debug logging is enabled, still fetch everything in order to
provide indication of available/supported data to users, as instructed
in docs.

balloob committed:
Fix tests

tmd224 committed:
Add Ambient Weather PWS Sensor component (#18551)

* Adding ambient_station.py sensor and updating requirements file

* Cleaning up code and fixing flake8 warnings

* Fixing flake8 and pylint warnings

* Adding ambient_station.py sensor and updating requirements file

* Cleaning up code and fixing flake8 warnings

* Fixing flake8 and pylint warnings

* Fixing bug, things are working well now

* removing nosetests file

* Adding changes as requested in pull request #18551

* Updating with more change requests from PR code review

* Removing SCAN_INTERVAL from PLATFORM_SCHEMA

* Removing unused import

* Adding platform schema validation for monitored_conditions and conf_units

* Updating link to documentation in doc-string.  File already named in doc repo

* Only setup platform if component can successfully communicate with API

* Inverting check for platform setup success

cgarwood committed:
Add note to issue template regarding frontend issues (#19295)

mezz64 committed:
Bump pyEight to fix Single Sleeper (#19316)

emontnemery committed:
Merge pull request #19223 from emontnemery/mqtt_sensor_json_attributes_topic

Add JSON attribute topic to MQTT sensor

scop committed:
Upgrade pytest to 4.0.2 (#19327)

scop committed:
Upgrade pydocstyle to 3.0.0 (#19328)

exxamalte committed:
bump georss_client to 0.5 (#19337)

chrillux committed:
Add Brottsplatskartan sensor (#19018)

* Added brottsplatskartan sensor

* Update brottsplatskartan sensor

* Remove redundant configuration checks.
* Set attributes during initialization.
* Setup brottsplatskartan module in setup_platform().
* Just do a simple return, no need for returning false.

* Remove file that was used during testing

* Better variable name and remove excessive newline.

* More updates to brottsplatskartan

* Import uuid at the top of the module.
* Use sensor as domain.
* Don't fire custom events.
* Remove period from logging message.

* Adding validation for area parameter

* Validate empty area configuration

* Fixing indentation.

* Fixing the config schema

emontnemery committed:
Fix broken sensor.mqtt json_attributes deprecation message (#19349)

jensihnow committed:
Adds io:OnOffIOComponent as switch to tahoma.py (#19338)

fredrike committed:
Add config flow for Daikin (#19182)

* config flow for daikin

* tox test

* return fixes

* tox test fixes

* tox formatting

glentakahashi committed:
Fix not being able to update entities (#19344)

When editing an entity in the frontend dialog, pressing save causes a "save failed: Entity is already registered" error. This is because the frontend always sets `name` and `new_entity_id` in the websocket command even if they haven't been changed. This adds a check that the `new_entity_id` is actually different from `entity_id` before erroring that the `new_entity_id` is already registered.

ludeeus committed:
Add traccar motion, speed and battery_level attributes (#19090)

* Added motion, speed and battery attributes.

* Use dict[key] when we know there is a value there.

Co-Authored-By: ludeeus <joasoe@gmail.com>

* Use dict[key] when we know there is a value there.

Co-Authored-By: ludeeus <joasoe@gmail.com>

* Use dict[key] when we know there is a value there.

Co-Authored-By: ludeeus <joasoe@gmail.com>

* Use dict[key] when we know there is a value there.

Co-Authored-By: ludeeus <joasoe@gmail.com>

* Use dict[key] when we know there is a value there.

Co-Authored-By: ludeeus <joasoe@gmail.com>

* Use dict[key] when we know there is a value there.

Co-Authored-By: ludeeus <joasoe@gmail.com>

PeteBa committed:
Initialise plant attributes at startup (#19315)

* Initialise plant attributes at startup

* Pvizeli review comments

* Martin review change

amelchio committed:
Remove recorder purge protection (#19358)

OttoWinter committed:
Add native ESPHome API component (#19334)

* Create esphomelib component

* Update requirements

* Remove python 2 string literals

* MVP

* Remove config flow

* Remove config flow translations

* Use dispatcher more

* Use data classes

* Type Hints

* Cleanup on remove

* Use helper and cleanup

* Fix HA stop listener cleanup

* Add config flow

* Fix SyntaxError for Py3.5

* Update

* Lint

* Re-do tests

* :sparkles: Rename to ESPHome

* Better error message for resolve errors

* Fix tests when aioesphomeapi not installed

* Refactor mock

* Update requirements

* Add strings.json

* :tea: 100% config flow  test coverage

bachya committed:
Add support for statewide data for Flu Near You (#19341)

* Initial changes

* Add support for state-wide data for Flu Near You

* Bumped requirements

* Member comments

pvizeli committed:
Use unicode slugify (#19192)

* Update __init__.py

* Update setup.py

* Update requirements_all.txt

* Update __init__.py

* Update __init__.py

* Update __init__.py

* Update __init__.py

* Update __init__.py

* remove `-`

* fix packages

* Update package_constraints.txt

* Update __init__.py

* Update package_constraints.txt

* Update requirements_all.txt

* Update setup.py

* Fix tests

* Fix line issue

* fix all test

* fix type

* Fix lint

fabaff committed:
Upgrade youtube_dl to 2018.12.17 (#19378)

fabaff committed:
Upgrade TwitterAPI to 2.5.7 (#19382)

fabaff committed:
Upgrade ruamel.yaml to 0.15.81 (#19384)

fredrike committed:
Version bump pydaikin (#19388)

* version bump pydaikin

* remove requirements from platform

fabaff committed:
Upgrade keyrings.alt==3.1.1 (#19386)

fabaff committed:
Set pytz to >=2018.07 (#19387)

marchingphoenix reviewed (approved):
Looks good to me. Thanks!

fabaff committed:
Upgrade RPi.GPIO to 0.6.5 (#19392)

fabaff committed:
Add air pollutants PROP_TO_ATTR (#19336)

* Add PROP_TO_ATTR

* Change condition

fredrike committed:
fix unique_id for Tellduslive sensors (#19389)

OttoWinter committed:
Add native ESPHome API binary sensor (#19371)

* Add esphomelib native API binary sensor

* Fixes

* :ambulance: Lint

OttoWinter committed:
Add native ESPHome API cover (#19373)

* Add esphomelib native API cover

* Update

* :ambulance: Lint

OttoWinter committed:
Add native ESPHome API fan (#19374)

* Add esphomelib native API fan

* Update

* :ambulance: Lint

OttoWinter committed:
Add native ESPHome API light (#19375)

* Add esphomelib native API light

* Update

* :ambulance: Lint

OttoWinter committed:
Add native ESPHome API switch (#19376)

* Add esphomelib native API switch

* Update

* :ambulance: Lint

OttoWinter committed:
Add native ESPHome API text sensor (#19377)

* Update

* :ambulance: Lint

OttoWinter committed:
ESPHome Native API Restore Entities on startup (#19379)

* Update __init__.py

* Use attr.fields_dict

OttoWinter committed:
Add native ESPHome API device registry feature (#19381)

* Add native ESPHome API device registry feature

* :sweat_smile: Actually call method

* Run script/gen_requirements_all

* Don't prefix sw_version

OttoWinter committed:
Add native ESPHome API service call feature (#19401)

* Add native ESPHome API service call feature

* :ambulance: Fix

balloob commented on homeassistant/components/sensor/islamic_prayer_times.py:
```diff
@@ -0,0 +1,235 @@
+    async def async_fire_prayer_event(self):
+        """Fire event for respective prayer time."""
+        _LOGGER.debug("Firing Event for: %s", self.sensor_type)
+        self.hass.bus.async_fire('islamic_prayer_time', {'prayer':
```
Entities are not allowed to fire events.

uchagani replied:
hmm okay i'll remove this.

balloob commented on homeassistant/components/sensor/islamic_prayer_times.py:
```diff
@@ -0,0 +1,235 @@
+            _LOGGER.debug("Creating event trigger for: %s", self._name)
+
+            async_track_point_in_time(self.hass,
+                                      self.async_fire_prayer_event(),
```
This makes no sense to call the function and then pass it as callback

uchagani replied:
this was to fire an event whenever a specific time happened.  but since we're not firing events anymore i'll remove this.

balloob commented on homeassistant/components/sensor/islamic_prayer_times.py:
```diff
@@ -0,0 +1,235 @@
+        """Update the sensor."""
+        prayer_time = self.prayer_times_data.prayer_times[self.name]
+        pt_dt = self.get_prayer_time_as_dt(prayer_time)
+        self._state = pt_dt.strftime(self._display_format)
```
state should be a iso formatted datetime stamp. Then the device class should be set to "timestamp".

uchagani replied:
good point, i'll update this.

fabaff committed:
Upgade colorlog to 4.0.2 (#19390)

MartinHjelmare commented on homeassistant/components/sensor/islamic_prayer_times.py:
```diff
@@ -0,0 +1,235 @@
+        prayer_time = self.prayer_times_data.prayer_times[self._name]
+        pt_dt = self.get_prayer_time_as_dt(prayer_time)
+        self._state = pt_dt.strftime(self._display_format)
+        _LOGGER.debug("%s State: %s", self.sensor_type, self._state)
```
We don't want side effects like logging in init method.

sdague committed:
Provide charging indicator for mychevy (#19348)

* Provide charging indicator for mychevy

This expands the mychevy sensor for the battery level to also know if
the system is currently charging to get the correct icon.

* address review feedback

MartinHjelmare commented on tests/components/sensor/test_islamic_prayer_times.py:
```diff
@@ -0,0 +1,112 @@
+                "Midnight": "00:45"}
+
+
+class TestIslamicPrayerTimesSensor():
```
We prefer standalone pytest test functions for new tests. See the deconz test suite for example.

eliseomartelli committed:
Add GTT Sensor (#18449)

* added gtt sensor

* removed trailing space

* updated requirements_all

* fixed two errors in the code style

* fixed imperative in docstring

* disabled pylint false positive

* fixed description on top of the file

* added files to .coveragerc

* fixes

* linting

* fixed linting 👍

* left a trailing space, now it's gone.

* fix

molobrakos committed:
Upgrade volvooncall to 0.8.7 (#19398)

deftdawg committed:
Cast lametric cycles parameter to int (#19370)

c-soft committed:
Satel integra monitor outputs (#19149)

* Adjusted api for new library version.

* Added support for output monitoring. Added initial status check for the alarm.

* Added default values to the configuration as per review notes.

mretegan committed:
Fix cpu_temp issue on Raspberry Pi (#19404)

nhorvath committed:
Add camera selection config to skybell camera (#19310)

* add camera selection config to skybell camera

* code review changes.

mezz64 committed:
Bump pyEmby to 1.6, add channel media type mapping (#19318)

Commit:
update platform based on feedback

Commit:
update platform based on feedback

uchagani commented:
Made changes as requested:

@balloob I removed the event-related code and updated the states to store ISO formatted datetime stamp along with adding the timestamp device class property.

@MartinHjelmare I removed the logging from init and updated the tests to the standalone pytest functions.

fabaff committed:
Add openSenseMap air pollutants platform (#19357)

* Add openSenseMap air pollutants platform

* Fix issues (Docstring, log entries and check)

* Use SCAN_INTERVAL and name handling

MartinHjelmare commented on tests/components/sensor/test_islamic_prayer_times.py:
```diff
@@ -0,0 +1,118 @@
+    """Test Islamic prayer times sensor creation."""
+    with MockDependency('prayer_times_calculator') as mock_pt_calc:
+        mock_pt_calc.prayer_times = PRAYER_TIMES
+        pt_sensor = IslamicPrayerTimeSensor('fajr', mock_pt_calc)
```
Don't test the entity in isolation. Set up the corresponding component as in the first tests, let the platform add the entity, then fire a mock time event while patching time to have the update be called for the entity. Then assert the state of the entity by getting it from the core state machine.

MartinHjelmare replied:
See the manual alarm_control_panel platform tests for example.

balloob committed:
Remove reviewed by hound. That's not worth a badge.

rohankapoorcom committed:
Optionally disable ssl verification for mjpeg (#19277)

mopolus committed:
Fix IHC config schema (#19415)

* Update __init__.py

Update "unit" -> "unit_of_measurement" and configuration (from plural to singular)

* Update __init__.py

* Removing vol.ALLOW_EXTRA arguments

* Update __init__.py

fabaff committed:
Use string foratting (#19427)

eliseomartelli committed:
Add Prezzibenzina (Italian Fuel Price) Sensor (#19297)

* complete(?)

* fixed linting

* update requirements

* added to coveragerc

* fixed linting

* added ability to set custom name

* fixed linting

* added filter

* spacing

* Added list of possible fuels

* Minor updates

fabaff committed:
Upgrade sqlalchemy to 1.2.15 (#19383)

fabaff committed:
Upgrade requests to 2.21.0 (#19385)

OttoWinter committed:
Miscellaneous ESPHome cleanups (#19425)

fabaff committed:
Fix typo (#19433)

Commit:
update tests not test platform in insolation

Commit:
update requirements file

uchagani commented:
@MartinHjelmare looks like i messed up rebasing this against dev.  I'm going to close out this PR and open a new clean one.
